src/absil/ilwrite.fs

```diff
@@ -3875,7 +3874,28 @@ let writeDirectory os dict =
 
 let writeBytes (os: BinaryWriter) (chunk:byte[]) = os.Write(chunk,0,chunk.Length)  
 
-let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer: ILStrongNameSigner option, fixupOverlappingSequencePoints, emitTailcalls, showTimes, dumpDebugInfo) modul noDebugData =
+type EmitStreamProvider = Lazy<Choice<EmitTo, Diagnostic>>
+and EmitTo =
+    | EmittedFile of string
+    | EmittedStream of Stream
+and Diagnostic = string
+
+
+// Either
+let (|Success|Failure|) = function
```

Comment:
Please just keep Choice1Of2/Choice2Of2 and remove the active pattern.  That's in keeping with enough of the style of the rest of the compiler and this PR isn't the right place to change the style. THanks